modules/nw-networkpolicy-cross-namespace-communication.adoc

```diff
@@ -0,0 +1,127 @@
+// Module included in the following assemblies:
+//
+// * networking/network_security/network_policy/nw-networkpolicy-full-multitenant-isolation.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="nw-networkpolicy-cross-namespace-communication_{context}"]
+= Creating a network policy for cross-namespace communication
+
+To allow pod-to-pod communication across namespaces, you must create a label for the primary namespace and add a `namespaceSelector` query and a `podSelector` query. 
+
+.Prerequisites
+
+* You have created the `deny-by-default` network policy and applied it to all necessary namespaces.
+
+.Procedure
+
+. Create the following `allow-n1-a-to-n2-b` network policy to allow pods across namespaces to communicate with each other. With this YAML, pods in the `project-a` that are labeled with `send-data` can communicate with pods in the `project-b` namespace that are labeled with `receive-data`. The namespaces must also be labeled to allow for communication. Save the YAML in the `allow-n1-a-to-n2-b` file:
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-n1-a-to-n2-b
+spec:
+  podSelector:
+    matchLabels:
+      app: receive-data <1>
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          networking/namespace: n1 <2>
+      podSelector:
+        matchLabels:
+          app: send-data  <3>
+----
+<1> Apply the `app: receive-data` label to pods in the `project-b` namespace.
+<2> Apply the `n1` label to the `project-a` namespace.
+<3> Apply the `app: send-data` label to pods in the `project-a` namespace.
+
+. Apply the `allow-n1-a-to-n2-b` network policy to the `project-b` namespace by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f allow-n1-a-to-n2-b.yaml -n project-b
+----
+
+. Label the `project-a` namespace with the `networking/namespace=n1` label by entering the following command:
++
+[source,terminal]
+----
+$ oc label namespace project-a networking/namespace=n1
+----
+
+. Label the `project-b` namespace with the `networking/namespace=n2` label by entering the following command:
++
+[source,terminal]
+----
+$ oc label namespace project-b networking/namespace=n2
```

Comment:
This label is not needed as we do not have any reference to namespace selector in network policy.